src/Parser.y

```diff
@@ -330,6 +332,7 @@ Atom :: { Expr Parsed }
      | hole                                   { withPos1 $1 (Hole (Name (getHole $1))) }
      | '_'                                    { withPos1 $1 (Hole (Name (T.singleton '_'))) }
      | begin List1(CompStmt, ExprSep) end     { withPos2 $1 $3 $ DoExpr bindVar $2 }
+     | '(|' ListE1(PreAtom) '|)'              { withPos2 $1 $3 $ Idiom pureVar apVar $2 }
```

Comment:
For consistency, possibly should be `Expr0 ListE(PreAtom)` ?

Reply:
W-why?

Reply:
oh, for consistency with application? sure